lib/coverband/collectors/view_tracker.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require 'singleton'
+
+module Coverband
+  module Collectors
+    ###
+    # This class tracks view file usage via ActiveSupport::Notifications
+    #
+    # This is a port of Flatfoot, a project I open sourced years ago, but am now rolling into Coverband
+    # https://github.com/livingsocial/flatfoot
+    #
+    # TODO: test and ensure slim, haml, and other support
+    ###
+    class ViewTracker
+      DEFAULT_TARGET = Dir.glob('app/views/**/*.html.erb').reject { |file| file.match(/(_mailer)/) }
+      attr_accessor :target, :logged_views, :views_to_record
+      attr_reader :logger, :roots, :store, :ignore_patterns
+
+      def initialize(options = {})
+        raise NotImplementedError, 'View Tracker requires Rails 4 or greater' unless self.class.supported_version?
+
+        @project_directory = File.expand_path(Coverband.configuration.root)
+        @ignore_patterns = Coverband.configuration.ignore
+        @store = options.fetch(:store) { Coverband.configuration.store }
+        @logger = options.fetch(:logger) { Coverband.configuration.logger }
+        @target = options.fetch(:target) { DEFAULT_TARGET }
+
+        @roots = options.fetch(:roots) { Coverband.configuration.all_root_patterns }
+        @roots = @roots.split(',') if @roots.is_a?(String)
+
+        @logged_views = []
+        @views_to_record = []
```

Comment:
I beg your pardon, but may I ask why an array is a better choice here than a `Set`?

Reply:
Yeah, a set would make more sense and would be faster. That is a good idea, thanks @hosamaly, feel free to send a PR or I can add it to my feature list backlog.

Reply:
I opened PR #418